cardano-testnet/src/Test/Assert.hs

```diff
@@ -86,10 +85,11 @@ data TraceNodeIsLeader = TraceNodeIsLeader
   } deriving (Eq, Show)
 
 instance FromJSON TraceNodeIsLeader where
-  parseJSON = Aeson.withObject "TraceNodeIsLeader" $ \v ->
-    TraceNodeIsLeader
-      <$> v .: "kind"
-      <*> v .: "slot"
+  parseJSON = Aeson.withObject "TraceNodeIsLeader" $ \v -> do
+    k <- v .: "val" >>= (.: "kind")
+    if k == "TraceNodeIsLeader"
+    then TraceNodeIsLeader k <$> (v .: "val" >>= (.: "slot"))
+    else fail "Not the right kind"
```

Comment:
@Jasagredo @newhoggy What is the failure "Not the right kind" going to indicate to the user? What is not the right kind? The error message says pretty much nothing. 

@Jasagredo can you improve this message to something like `"Expected TraceNodeIsLeader but got: " <> k`

Reply:
Sure thing. This is only used in tests though, but yeah, a more informative error message would be better.